calculator_test.go

```diff
@@ -3,189 +3,427 @@ package calculator_test
+			a:    5.5,
+			b:    1.1,
+			nums: []float64{2.2},
+			want: 13.31,
 		},
 	}
 
-	for _, test := range successTestCases {
-		got := calculator.Multiply(test.a, test.b)
-		if !almostEqual(test.want, got) {
-			t.Errorf("Test Multiple(%f, %f) %s failed: want %f, got %f",
-				test.a, test.b, test.name, test.want, got)
+	for _, tc := range tcs {
+		got := calculator.Multiply(tc.a, tc.b, tc.nums...)
+		if !closeEnough(tc.want, got) {
+			t.Errorf("%s: Multiply(%f, %f, %v) : want %f, got %f",
+				tc.name, tc.a, tc.b, tc.nums, tc.want, got)
 		}
 	}
-
 }
 
 func TestDivide(t *testing.T) {
 	t.Parallel()
-	var successTestCases = []struct {
+	tcs := []struct {
 		name        string
-		want        float64
 		a           float64
 		b           float64
+		nums        []float64
+		want        float64
 		errExpected bool
 	}{
 		{
 			name:        "Two positive numbers",
-			want:        5,
 			a:           10,
 			b:           2,
+			want:        5,
 			errExpected: false,
 		},
 		{
 			name:        "Zero divided with any number",
-			want:        0,
 			a:           0,
 			b:           2,
+			want:        0,
 			errExpected: false,
 		},
 		{
 			name:        "Positive and negative number",
-			want:        -1.25,
 			a:           -5,
 			b:           4,
+			want:        -1.25,
 			errExpected: false,
 		},
 		{
 			name:        "Two negative numbers",
-			want:        4,
 			a:           -8,
 			b:           -2,
+			want:        4,
 			errExpected: false,
 		},
 		{
 			name:        "Two fractional numbers",
-			want:        2.75,
 			a:           3.3,
 			b:           1.2,
+			want:        2.75,
 			errExpected: false,
 		},
 		{
 			name:        "Divide by zero",
-			want:        0,
 			a:           2,
 			b:           0,
+			want:        0,
+			errExpected: true,
+		},
+		{
+			name:        "More than 2 numbers",
+			a:           40,
+			b:           5,
+			nums:        []float64{2, 2},
+			want:        2,
+			errExpected: false,
+		},
+		{
+			name:        "More than 2 fractions",
+			a:           5.5,
+			b:           1.1,
+			nums:        []float64{0.5, 0.2, 2.5},
+			want:        20,
+			errExpected: false,
+		},
+		{
+			name:        "Divide by 0 in slice",
+			a:           20,
+			b:           2,
+			nums:        []float64{2, 0},
+			want:        0,
 			errExpected: true,
 		},
 	}
 
-	for _, test := range successTestCases {
-		got, err := calculator.Divide(test.a, test.b)
+	for _, tc := range tcs {
+		got, err := calculator.Divide(tc.a, tc.b, tc.nums...)
 		errReceived := err != nil
-		if test.errExpected != errReceived{
-				t.Errorf("Test Divide (%f, %f) %s failed: Unexpected error received",
-					test.a, test.b, test.name)
-		} else {
-			if !test.errExpected && !almostEqual(test.want, got) {
-				t.Errorf("Test Divide (%f, %f) %s failed: want %f, got %f",
-					test.a, test.b, test.name, test.want, got)
-			}
+		if tc.errExpected != errReceived {
+			t.Errorf("%s: Divide (%f, %f, %v) : Unexpected error received",
```

Comment:
```suggestion
			t.Errorf("%s: Divide (%f, %f, %v) : Unexpected error status: %v",
```

In the case where we expected error but received `nil`, this isn't an unexpected error _received_, it's an unexpected error _status_ (that is, `nil`).